Gradle file example from Getting Started Guide does not work

**The context**

The guide gives an example of a gradle build file here: https://yidongnan.github.io/grpc-spring-boot-starter/en/server/getting-started.html

**The question**

However the resolution of that file files for the first dependency tasks

**Stacktraces and logs**

```gradle
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Build file '/home/xetra11/Development/projects/user-service/grpc-lib/build.gradle' line: 4

* What went wrong:
A problem occurred evaluating project ':grpc-lib'.
> Could not find method implementation() for arguments [io.grpc:grpc-protobuf:1.35.0] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':grpc-lib'.
> The com.google.protobuf plugin could not be applied during project evaluation. The Java plugin or one of the Android plugins must be applied to the project first.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 94ms

```

**The application's environment**

The build file:

```gradle
apply plugin: 'com.google.protobuf'

dependencies {
    compile "io.grpc:grpc-protobuf:1.35.0"
    compile "io.grpc:grpc-stub:1.35.0"
}

buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.14'
    }
}

protobuf {
    protoc {
        artifact = "com.google.protobuf:protoc:3.14.0"
    }
    generatedFilesBaseDir = "$projectDir/src/generated"
    clean {
        delete generatedFilesBaseDir
    }
    plugins {
        grpc {
            artifact = "io.grpc:protoc-gen-grpc-java"
        }
    }
    generateProtoTasks {
        all()*.plugins {
            grpc {}
        }
    }
}

buildscript {
    dependencies {
        classpath "com.google.protobuf:protobuf-gradle-plugin:0.8.14"
    }
}
```

@ST-DDT 
Thanks! Looks promising. Just one last thing is happening

```
3:46:07 PM: Executing task ':grpc-lib:build'...

> Task :grpc-lib:extractIncludeProto
> Task :grpc-lib:extractProto
> Task :grpc-lib:generateProto

> Task :grpc-lib:compileJava FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings
4 actionable tasks: 4 executed
/home/xetra11/Development/projects/user-service/grpc-lib/src/generated/main/grpc/net/devh/boot/grpc/examples/lib/SimpleGrpc.java:10: error: cannot find symbol
@javax.annotation.Generated(
                 ^
  symbol:   class Generated
  location: package javax.annotation
1 error

FAILURE: Build failed with an exception.
```

--- a/docs/en/server/getting-started.md
+++ b/docs/en/server/getting-started.md
@@ -62,10 +62,11 @@ We recommend splitting your project into 2-3 separate modules.
             <version>${grpc.version}</version>
         </dependency>
         <dependency>
-            <!-- Java 9+ compatibility -->
+            <!-- Java 9+ compatibility - Do NOT update to 2.0.0 -->
             <groupId>jakarta.annotation</groupId>
             <artifactId>jakarta.annotation-api</artifactId>
-            <version>2.0.0</version>
+            <version>1.3.5</version>
+            <optional>true</optional>
         </dependency>
     </dependencies>
 
@@ -124,7 +125,7 @@ repositories {
 dependencies {
     implementation "io.grpc:grpc-protobuf:${grpcVersion}"
     implementation "io.grpc:grpc-stub:${grpcVersion}"
-    implementation 'jakarta.annotation:jakarta.annotation-api:2.0.0' // Java 9+ compatibility
+    compileOnly 'jakarta.annotation:jakarta.annotation-api:1.3.5' // Java 9+ compatibility - Do NOT update to 2.0.0 (Compile only)
 }
 
 protobuf {

--- a/docs/en/server/getting-started.md
+++ b/docs/en/server/getting-started.md
@@ -125,7 +125,7 @@ repositories {
 dependencies {
     implementation "io.grpc:grpc-protobuf:${grpcVersion}"
     implementation "io.grpc:grpc-stub:${grpcVersion}"
-    compileOnly 'jakarta.annotation:jakarta.annotation-api:1.3.5' // Java 9+ compatibility - Do NOT update to 2.0.0 (Compile only)
+    compileOnly 'jakarta.annotation:jakarta.annotation-api:1.3.5' // Java 9+ compatibility - Do NOT update to 2.0.0
 }
 
 protobuf {

--- a/docs/en/server/getting-started.md
+++ b/docs/en/server/getting-started.md
@@ -44,19 +44,28 @@ We recommend splitting your project into 2-3 separate modules.
 #### Maven (Interface)
 
 ````xml
+    <properties>
+        <protobuf.version>3.14.0</protobuf.version>
+        <protobuf-plugin.version>0.6.1</protobuf-plugin.version>
+        <grpc.version>1.35.0</grpc.version>
+    </properties>
+
     <dependencies>
         <dependency>
             <groupId>io.grpc</groupId>
             <artifactId>grpc-stub</artifactId>
+            <version>${grpc.version}</version>
         </dependency>
         <dependency>
             <groupId>io.grpc</groupId>
             <artifactId>grpc-protobuf</artifactId>
+            <version>${grpc.version}</version>
         </dependency>
         <dependency>
             <!-- Java 9+ compatibility -->
             <groupId>javax.annotation</groupId>
             <artifactId>javax.annotation-api</artifactId>
+            <version>2.0.0</version>
         </dependency>
     </dependencies>
 
@@ -65,6 +74,7 @@ We recommend splitting your project into 2-3 separate modules.
             <extension>
                 <groupId>kr.motd.maven</groupId>
                 <artifactId>os-maven-plugin</artifactId>
+                <version>1.6.2</version>
             </extension>
         </extensions>
 
@@ -72,6 +82,7 @@ We recommend splitting your project into 2-3 separate modules.
             <plugin>
                 <groupId>org.xolstice.maven.plugins</groupId>
                 <artifactId>protobuf-maven-plugin</artifactId>
+                <version>${protobuf-plugin.version}</version>
                 <configuration>
                     <protocArtifact>com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}</protocArtifact>
                     <pluginId>grpc-java</pluginId>
@@ -93,11 +104,27 @@ We recommend splitting your project into 2-3 separate modules.
 #### Gradle (Interface)
 
 ````gradle
-apply plugin: 'com.google.protobuf'
+buildscript {
+    ext {
+        protobufVersion = '3.14.0'
+        protobufPluginVersion = '0.8.14'
+        grpcVersion = '1.35.0'
+    }
+}
+
+plugins {
+    id 'java-library'
+    id 'com.google.protobuf' version "${protobufPluginVersion}"
+}
+
+repositories {
+    mavenCentral()
+}
 
 dependencies {
-    compile "io.grpc:grpc-protobuf"
-    compile "io.grpc:grpc-stub"
+    implementation "io.grpc:grpc-protobuf:${grpcVersion}"
+    implementation "io.grpc:grpc-stub:${grpcVersion}"
+    implementation 'jakarta.annotation:jakarta.annotation-api:2.0.0' // Java 9+ compatibility
 }
 
 protobuf {
@@ -110,19 +137,13 @@ protobuf {
     }
     plugins {
         grpc {
-            artifact = "io.grpc:protoc-gen-grpc-java"
+            artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}"
         }
     }
     generateProtoTasks {
         all()*.plugins {
             grpc {}
         }
-    }
-}
-
-buildscript {
-    dependencies {
-        classpath "com.google.protobuf:protobuf-gradle-plugin:${protobufGradlePluginVersion}"
     }
 }
 

--- a/docs/en/server/getting-started.md
+++ b/docs/en/server/getting-started.md
@@ -63,8 +63,8 @@ We recommend splitting your project into 2-3 separate modules.
         </dependency>
         <dependency>
             <!-- Java 9+ compatibility -->
-            <groupId>javax.annotation</groupId>
-            <artifactId>javax.annotation-api</artifactId>
+            <groupId>jakarta.annotation</groupId>
+            <artifactId>jakarta.annotation-api</artifactId>
             <version>2.0.0</version>
         </dependency>
     </dependencies>

